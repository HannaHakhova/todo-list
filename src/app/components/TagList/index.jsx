import React from 'react';
import {List} from 'semantic-ui-react';
import {func, arrayOf, number, string, shape} from 'prop-types';
import Tag from 'components/Tag';


const TagList = ({tags, handleDelete}) => {
    const tagArray = tags.map(tag =>
        <Tag tag={tag} key={tag.id} handleDelete={handleDelete} />);
    return (
        <List>
            {tagArray}
        </List>
    );
};

TagList.propTypes = {
    handleDelete: func.isRequired,
    tags: arrayOf(shape({
        id: string.isRequired,
        title: string.isRequired,
        count: number.isRequired
    }).isRequired).isRequired
};

export default TagList;