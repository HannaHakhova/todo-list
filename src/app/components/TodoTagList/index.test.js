/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import {shallow} from 'enzyme';
import TagList from './index';

const wrapper = shallow(<TagList
    handleDelete={() => {}}
    tags={[{
        id: '1',
        title: 'Tag1'
    },
    {
        id: '2',
        title: 'Tag2'
    },
    {
        id: '3',
        title: 'Tag3'
    }]} />);

describe('TagList component', () => {
    it('should match expected snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
});