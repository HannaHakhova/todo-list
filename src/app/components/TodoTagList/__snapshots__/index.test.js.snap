// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagList component should match expected snapshot 1`] = `
<List>
  <Tag
    handleDelete={[Function]}
    key="1"
    tag={
      Object {
        "id": "1",
        "title": "Tag1",
      }
    }
  />
  <Tag
    handleDelete={[Function]}
    key="2"
    tag={
      Object {
        "id": "2",
        "title": "Tag2",
      }
    }
  />
  <Tag
    handleDelete={[Function]}
    key="3"
    tag={
      Object {
        "id": "3",
        "title": "Tag3",
      }
    }
  />
</List>
`;
